typescript-definitions-derive/src/tests.rs

```diff
@@ -3,108 +3,285 @@ use super::Typescriptify;
-        match result {
-            Ok(_x) => assert!(false, "expecting panic!"),
-            Err(ref msg) => assert_snapshot_matches!( msg.downcast_ref::<String>().unwrap(),
-            @r###"S: guard must be true or false not ""blah"""###
-            ),
-        }
+        assert_conversion!(
+            tokens,
+            "export type SimpleEnum = \n | { t: \"Foo\" } \n | { t: \"Bar\" }"
+        )
     }
-    /*
+
     #[test]
-    fn turbofish() {
+    fn enum_with_struct_variants_is_converted() {
         let tokens = quote!(
-            #[derive(TypeScriptify)]
-            #[ts(turbofish = "<i32>")]
-            struct S<T> {
-                a: i32,
-                b: Vec<T>,
+            #[serde(tag = "t", content = "c")]
+            enum ComplexEnum {
+                Foo(String),
+                Bar { Baz: i32 },
             }
         );
-        let ty = Typescriptify::parse(true, tokens);
-        let i = &ty.ctxt.ident;
-        let g = ty.ctxt.global_attrs.turbofish.unwrap_or_else(|| quote!());
-        let res = quote!(#i#g::type_script_ify()).to_string();
-        assert_snapshot_matches!(res,
-        @"S < i32 > :: type_script_ify ( )" );
-    }
-    */
+        assert_conversion!(
+            tokens, "export type ComplexEnum = \n | { t: \"Foo\"; c: string } \n | { t: \"Bar\"; c: { Baz: number } }"
+        )
+    }
+
     #[test]
-    fn bad_ts_as() {
+    fn enum_with_complex_inner_types_is_converted() {
         let tokens = quote!(
-            #[derive(TypeScriptify)]
-
-            struct S<T> {
-                #[ts(ts_as = "😀i32>")]
-                a: i32,
-                #[ts(ts_as = "T[]")]
-                b: Vec<T>,
+            #[serde(tag = "t", content = "c")]
+            enum ComplexEnum {
+                Foo(Vec<Bar>),
+                Bar(Option<Baz>),
             }
         );
-        let result = std::panic::catch_unwind(move || Typescriptify::new(tokens).parse(true));
+        assert_conversion!(
+            tokens, "export type ComplexEnum = \n | { t: \"Foo\"; c: Bar [] } \n | { t: \"Bar\"; c: Baz | null }"
+        )
+    }
+
+    // Error tests
+
+    #[test]
+    fn conversion_is_only_valid_for_structs_or_enums() {
+        let tokens = quote!(type Foo(String));
+
+        let result = std::panic::catch_unwind(move || Typescriptify::new(tokens).parse());
         match result {
-            Ok(_x) => assert!(false, "expecting panic!"),
-            Err(ref msg) => assert_snapshot_matches!( msg.downcast_ref::<String>().unwrap(),
-            @r###"2 errors:
-	# ts_as: "😀i32>" is not a valid rust type
-	# ts_as: "T[]" is not a valid rust type"###
-            ),
+            Ok(_) => panic!("expecting panic!"),
+            Err(ref msg) => assert!(msg
+                .downcast_ref::<String>()
+                .unwrap()
+                .contains("expected one of: `struct`, `enum`, `union`")),
         }
     }
+
```

Comment:
None of these compiled at all previously, nevermind passing, and it's less bang for buck fixing it than getting the type conversion test coverage. We should probably bring them back in future though.